staging/src/k8s.io/api/discovery/v1/types.go

```diff
@@ -40,7 +40,7 @@ type EndpointSlice struct {
 	// supported:
 	// * IPv4: Represents an IPv4 Address.
 	// * IPv6: Represents an IPv6 Address.
-	// * FQDN: Represents a Fully Qualified Domain Name.
+	// * FQDN: Represents a Fully Qualified Domain Name. [DEPRECATED]
```

Comment:
what does deprecated mean? also document this on the AddressTypeFQDN constant?

Reply:
can we be clearer about what deprecated means, like "this value is not set by Kubernetes controllers X, Y, Z, and has undefined semantics"?

Reply:
I actually wonder if we would want a - brief - KEP where we write out what the semantics for a deprecated variant are. Maybe a change to https://kubernetes.io/docs/reference/using-api/deprecation-policy/ as well.

It's a pain to do that, but: it's also a pain to end up adopting a de facto policy that we later come to regret.

Reply:
No core Kubernetes component creates slices of that type, and a slice with that type has no defined semantics beyond the fact that kube-proxy will ignore it.

Third-party components may be using it for implementation-defined behavior, and that in particular is what we want to deprecate, because it's possible different third-party components are using it in slightly different ways due to the lack of defined semantics.

Reply:
I think what we need to define is not what deprecated mean for this enum variant; instead, it's about what it means across the Kubernetes project overall when we do this kind of deprecation.

AIUI this PR has to settle that because it's the first time we're deprecating a variant of a GA enum-like-field, whilst leaving the remaining values valid.

Reply:
We might also need to update things like API docs generators. Of course, we can do the deprecation now and live with confusing docs just for this one field; the generator updates aren't an outright blocker for merging this PR.